src/MartinGeorgiev/Doctrine/DBAL/Types/ValueObject/Ltree.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MartinGeorgiev\Doctrine\DBAL\Types\ValueObject;
+
+use MartinGeorgiev\Doctrine\DBAL\Types\ValueObject\Exceptions\InvalidLtreeException;
+
+/**
+ * @since 3.5
+ *
+ * @author Pierre-Yves Landuré
+ *
+ * @phpstan-consistent-constructor
+ */
+class Ltree implements \Stringable, \JsonSerializable
+{
+    /**
+     * @param list<non-empty-string> $pathFromRoot A list with one element represents the root. The list may be empty.
+     *
+     * @throws InvalidLtreeException if the pathFromRoot is not a valid ltree path
+     *                               (contains labels which are empty or contains one or more dots)
+     */
+    public function __construct(
+        private readonly array $pathFromRoot,
+    ) {
+        self::assertListOfValidLtreeNodes($pathFromRoot);
+    }
+
+    #[\Override]
+    public function __toString(): string
+    {
+        return \implode('.', $this->pathFromRoot);
+    }
+
+    /**
+     * @throws InvalidLtreeException if $ltree contains invalid/empty labels (e.g., consecutive dots)
+     */
+    public static function fromString(string $ltree): static
+    {
+        if ('' === $ltree) {
+            return new static([]);
+        }
+
+        $pathFromRoot = \explode('.', $ltree);
+
+        return new static($pathFromRoot);
+    }
+
+    /**
+     * @return list<non-empty-string>
+     */
+    #[\Override]
+    public function jsonSerialize(): array
+    {
+        return $this->getPathFromRoot();
+    }
+
+    /**
+     * @return list<non-empty-string>
+     */
+    public function getPathFromRoot(): array
+    {
+        return $this->pathFromRoot;
+    }
+
+    /**
+     * @throws InvalidLtreeException if the ltree is empty
+     */
+    public function getParent(): static
+    {
+        if ($this->isEmpty()) {
+            throw InvalidLtreeException::forImpossibleLtree('Empty ltree has no parent.');
+        }
+
+        $parentPathFromRoot = \array_slice($this->pathFromRoot, 0, -1);
+
+        return new static($parentPathFromRoot);
+    }
+
+    public function isEmpty(): bool
```

Comment:
What's the purpose of having support for empty instances? It is not a logical tree in this case, is it?